targets/TARGET_STM/us_ticker_32b.c

```diff
@@ -23,33 +23,30 @@
 
 TIM_HandleTypeDef TimMasterHandle;
 
-static int us_ticker_inited = 0;
-
 void us_ticker_init(void)
 {
-    if (us_ticker_inited) return;
-    us_ticker_inited = 1;
-
-    TimMasterHandle.Instance = TIM_MST;
-
-    HAL_InitTick(0); // The passed value is not used
+    /* NOTE: assuming that HAL tick has already been initialized! */
 }
 
 uint32_t us_ticker_read()
 {
-    if (!us_ticker_inited) us_ticker_init();
     return TIM_MST->CNT;
 }
 
 void us_ticker_set_interrupt(timestamp_t timestamp)
 {
-    TimMasterHandle.Instance = TIM_MST;
+    /* Disable global IRQs */
+    core_util_critical_section_enter();
```

Comment:
> Well, I believe that us_ticker_set_interrupt() might also be called not from interrupt context and out of a critical section, e.g. see here.

HAL is not thread safe, thus a protection should happen above. As it is in mbed ticker implementation. This should be stated in tickers HAL API. 

> Calling also __HAL_TIM_DISABLE_IT(&TimMasterHandle, TIM_IT_CC1) and __HAL_TIM_ENABLE_IT(&TimMasterHandle, TIM_IT_CC1) seems to be redundant, though.

If we keep critical section then yes.

> Please refer to the diffs under Files changed, there has been an update.
 
I see now the update. It would be helpful to provide more information in the commit messages about changes (why we adding critical section to us ticker implementation , why to hal_sleep/deepsleep).